eng/common/scripts/Verify-AgentOS.ps1

```diff
@@ -0,0 +1,60 @@
+param (
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $AgentPool,
+
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $AgentImage
+)
+
+class Rule {
```

Comment:
What value does creating a custom class here provide?

Reply:
Just made it nicer when I dumped out a table of all the valid combinations.